ncm-afsclt/src/main/perl/afsclt.pm

```diff
@@ -2,997 +2,306 @@
-    my $ISA = '';
-    my $HWname = LC::Sysinfo::uname()->machine;
-    if ($HWname =~ /x86_64/) {
-      $ISA = '../../lib64/security';
+    if ( $config->elementExists("$mypath/thesecells") ) {
+        my $cells = $config->getElement("$mypath/thesecells")->getTree();
+        my $thesecells_fh = CAF::FileWriter->new( $THESECELLS, log => $self );
+        print $thesecells_fh join ( " ", @$cells ) . "\n";
+        if ( $thesecells_fh->close() ) {
+            $self->info("Configured cell list for authentication $THESECELLS");
+        }
     }
-    my $libpamtmp = $libpam;
-    $libpamtmp =~ s:^(.*)\$ISA/(.*):$1$ISA/$2:;
-    if(! -e "$libpamtmp") {
-      $self->warn("Cannot find $libpam (=$libpamtmp), will not configure PAM");
-      return 0;
+    elsif ( -f $THESECELLS ) {
+        if ($NoAction) {
+            $self->info("Would remove $THESECELLS");
+        }
+        else {
+            if ( unlink($THESECELLS) ) {
+                $self->info("Removed cell list for authentication $THESECELLS");
+            }
+            else {
+                $self->error("Could not remove $THESECELLS: $!");
+            }
+        }
     }
+}
 
-    my $pam_systemauth = "/etc/pam.d/system-auth";
-    my $pam_screensaver = '/etc/pam.d/screensaver-auth'; #optional
-
-    if ( ! -e "$pam_systemauth") {
-      $self->warn("Cannot find $pam_systemauth - not touching it");
-    } else {
-      $ret = LC::Check::file($pam_systemauth,
-			     backup => '.ncm_orig',
-			     source => $pam_systemauth,
-			     code => sub {  # gets actual, returns expected.
-				    my $actual = shift;
-				    $self->debug(4, "$pam_systemauth is now:\n$actual");
-				    my @actual = split (/\n/, $actual);
-				    my @expected = grep { $_ !~
-							   /(added by $compname|pam_afs|pam_krb5afs|pam_heimdal)/
-							 } @actual; # remove other AFS/Krb PAMs
-				    @expected = map {
-				      # add our lib after pam_unix
-				      if ($_ =~ /^(auth) (\s*)(\S+) (\s*)\S*pam_unix/) {
-					($_, "# next line added by $compname", "$1 $2$3 $4$libpam $libpam_options_auth_auth");
-				      } elsif ($_ =~ /^(session) (\s*)(\S+)(\s+)\S*pam_unix/) {
-					($_, "# next line added by $compname", "$1 $2$3 $4$libpam $libpam_options_auth_session");
-				      } elsif ($_ =~ /^(password) (\s*)(\S+)(\s+)\S*pam_unix/) {
-					($_, "# next line added by $compname", "$1 $2$3 $4$libpam $libpam_options_auth_passwd");
-				      } else {
-					$self->debug(5, "ignoring PAM line $_");
-					$_;
-				      }
-				    } @expected;
-				    my $expected = join("\n",@expected)."\n";
-				    $self->debug(4, "$pam_systemauth will be:\n$expected");
-				    return $expected;
-				   }
-			  );
+sub Configure_Config {
+    my ( $self, $config ) = @_;
+
+    # cache size is configured in a separate function
+    if ( $config->elementExists("$mypath/afsmount") ) {
+        my $new_afsmount = $config->getValue("$mypath/afsmount");
+        if ( $new_afsmount ne '/afs' ) {
```

Comment:
I had a closer look at this, it seems totally useless. This parameter is not used anywhere.

it wouldn't work for us anyway because our afs is not mounted to /afs.